src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -274,13 +275,36 @@ CHIP_ERROR DiscoveryImplPlatform::Advertise(const OperationalAdvertisingParamete
 
     ReturnErrorOnFailure(MakeInstanceName(service.mName, sizeof(service.mName), params.GetFabricId(), params.GetNodeId()));
     strncpy(service.mType, "_chip", sizeof(service.mType));
-    service.mProtocol      = MdnsServiceProtocol::kMdnsProtocolTcp;
-    service.mPort          = CHIP_PORT;
-    service.mTextEntries   = nullptr;
-    service.mTextEntrySize = 0;
-    service.mInterface     = INET_NULL_INTERFACEID;
-    service.mAddressType   = Inet::kIPAddressType_Any;
-    error                  = ChipMdnsPublishService(&service);
+    service.mProtocol = MdnsServiceProtocol::kMdnsProtocolTcp;
+    service.mPort     = CHIP_PORT;
+    TextEntry crmpRetryIntervalEntries[2];
```

Comment:
1. I suggest moving this part a few lines above so that all MdnsService members are initialized in a single place.
2. Note that according to the spec CRI/CRA should <= 1 hour. Could you add some `CRI = std::min(CRI, 1 hour)` etc.?
3. CRI/CRA should be converted to a string not passed as a binary value.
4. Can we set CRI/CRA only for SEDs?

Reply:
1,2,3 - right, I will change it. 4 - in that case you would like to leave the responsibility of using default CRMP intervals to controller?

Reply:
1,2 solved, got a problem with 3, as in CHIP device string looks good, but on the server side I see:
TXT: [CRI=32343130303000, CRA=323336333030] -> before each one number there is "3" digit added and there are two times as many characters, as it should be. E.g. 323336333030->236300 is correct.

Reply:
3. Problem seems to not be an issue, as those characters are just ASCII codes of digits.
4. Discussed offline to set also default CRMP values, at least until controller will start supporting using default values on his own.